packages/adminrouter/extra/src/includes/http/common.conf

```diff
@@ -1,9 +1,17 @@
-# Enable VTS module for metrics
-# Set the shared memory zone size to 32MB to be sure to
-# contain all metrics in the VTS module configuration.
-# If the VTS module configuration enables dynamic metric
-# keys this needs to be increased to accommodate for that.
-vhost_traffic_status_zone shared:vhost_traffic_status:32m;
+# Enable VTS module for metrics.
+# The shared memory zone size must be sufficiently large to ensure that all
+# metrics that could potentially be generated by the VTS module's
+# vhost_traffic_status_filter_by_set_key directive can be stored. The metrics
+# data scales with the number of metric nodes. A metric node in our case is a
+# set of data about one detail, such as the number of requests made with a
+# particular client, or the number of responses with a particular status code,
+# or the number of URIs requested from Admin Router. In theory, the number of
+# URIs can be infinitely large. If the shared memory zone is near 86% of the
+# declared maximum size the VTS module will stop creating new metric nodes.
+# Therefore, we risk missing out on data if the URIs requested grows large.
+# See DCOS_OSS-4677 for limiting the number of nodes and employing a LRU
+# mechanism.
+vhost_traffic_status_zone shared:vhost_traffic_status:8m;
```

Comment:
As discussed with @timaa2k, we should ensure that this is large enough to handle MWT sized clusters in a reasonable way. Tim suggests that 64MB is a reasonable value. If this causes problems with more constrained environments (e.g. minidcos on Docker/Vagrant) then we should look later at possibly adding a configuration variable for this, but also at other processes in DC/OS that use significantly more memory than Admin Router.

Reply:
Further testing revealed that 16MB is more appropriate than 64MB because until there is a dedicated Telegraf plugin for Admin Router that drops useless metrics generated in the process.
Any Prometheus server that scrapes Telegraf must download these within its scrape interval and timeout. At 9.2MB Admin Router metrics nodes this amounts to 53MB of time-series data for Telegraf. That's close to a factor of 5 of useless metrics. Going from 16MB to 64MB would mean approximately 50MB metric node data, thus 250MB to download from Telegraf. That would require an optimistic network connection of like 5MB/s. Once the Telegraf plugin is in place we could still tune the shared memory size.